lib/metanorma/site_manifest.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require "lutaml/model"
+
+module Metanorma
+  class SiteManifest < Lutaml::Model::Serializable
+    class SourceEntries < Lutaml::Model::Serializable
+      attribute :files, :string, collection: true
+    end
+
+    class SiteMetadata < Lutaml::Model::Serializable
+      attribute :organization, :string
+      attribute :name, :string
+    end
+
+    class SiteTemplate < Lutaml::Model::Serializable
+      attribute :path, :string
+      attribute :stylesheet, :string
+      attribute :output_filename, :string
+    end
+
+    class Manifest < Lutaml::Model::Serializable
+      attribute :source, SourceEntries
+      attribute :collection, SiteMetadata
+      attribute :template, SiteTemplate
+    end
+    attribute :metanorma, Manifest
```

Comment:
Wrapping a Class within a Class here is a bit weird, because each of these classes define a particular hierarchy. e.g. can you call the base the "Base" or the "Root" so SiteManifest is a module?